python/cuml/ensemble/randomforest.pyx

```diff
@@ -0,0 +1,306 @@
+        self.bootstrap_features = bootstrap_features
+        self.rows_sample = rows_sample
+        self.max_leaves = max_leaves
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.max_features = max_features
+        self.min_samples_split = min_samples_split
+        self.type_model = self._get_type(type_model)
+        self.bootstrap = bootstrap
+        self.verbose = False
+        self.n_bins = n_bins
+        self.rf_classifier32 = NULL
+        self.rf_classifier64 = NULL
+        self.n_rows = None
+        self.n_cols = None
+        self.preds = None
+
+    def _get_type(self, type_model):
+        if type_model == "classifier":
+            self.type_model = 1
+        else:
+            self.type_model = 0
+
+    def fit(self, X, y):
+
+        cdef uintptr_t X_ptr, y_ptr
+
+        if self.rf_classifier32 != NULL:
+            del self.rf_classifier32
+
+        if self.rf_classifier64 != NULL:
+            del self.rf_classifier64
+
+        X_m, X_ptr, self.n_rows, self.n_cols, self.dtype = \
+            input_to_dev_array(X, order='F')
+        y_m, y_ptr, _, _, _ = input_to_dev_array(y)
+
+        cdef cumlHandle * handle_ =\
+            <cumlHandle *> <size_t> self.handle.getHandle()
+
+        unique_labels = (cupy.unique(y)).__len__()
+
+        rf_param = set_rf_class_obj(<int> self.max_depth,
+                                    <int> self.max_leaves,
+                                    <float> self.max_features,
+                                    <int> self.n_bins,
+                                    <int> self.split_algo,
+                                    <int> self.min_rows_per_node,
+                                    <bool> self.bootstrap_features,
+                                    <bool> self.bootstrap,
+                                    <int> self.n_estimators,
+                                    <int> self.rows_sample)
+
+        self.rf_classifier32 = new \
+            rfClassifier[float](rf_param)
+        self.rf_classifier64 = new \
+            rfClassifier[double](rf_param)
+
+        if self.dtype == np.float32:
+            fit(handle_[0],
+                self.rf_classifier32,
+                <float*> X_ptr,
+                <int> self.n_rows,
+                <int> self.n_cols,
+                <int*> y_ptr,
+                <int> unique_labels)
+        else:
+            fit(handle_[0],
+                self.rf_classifier64,
+                <double *> X_ptr,
+                <int> self.n_rows,
+                <int> self.n_cols,
+                <int *> y_ptr,
+                <int> unique_labels)
+
+        # make sure that the `fit` is complete before the following delete
+        # call happens
+        self.handle.sync()
+        del(X_m)
+        del(y_m)
+        return self
+
+    def predict(self, X):
+
+        cdef uintptr_t X_ptr
+        X_ptr = X.ctypes.data
+        self.n_rows, self.n_cols = np.shape(X)
```

Comment:
You don’t want to override self.n_cols here, you want to check that n_cols of the input is the same as self.n_cols